python/python/lancedb/table.py

```diff
@@ -2464,7 +2464,31 @@ async def _do_merge(
         on_bad_vectors: str,
         fill_value: float,
     ):
-        pass
+        schema = await self.schema()
+        if on_bad_vectors is None:
+            on_bad_vectors = "error"
+        if fill_value is None:
+            fill_value = 0.0
+        data, _ = _sanitize_data(
+            new_data,
+            schema,
+            metadata=schema.metadata,
+            on_bad_vectors=on_bad_vectors,
+            fill_value=fill_value,
+        )
+        if isinstance(data, pa.Table):
```

Comment:
Does `_sanitize_data` ever return something that isn't `pa.Table`?

Reply:
I'm not sure. This is taken from `AsyncTable.add()`. I think right now it's always a table. It should really be better typed...